Give virtio_net driver (used in VCI) ethtool support for speed/duplex
so bonding driver can get slave interface speed/duplex when configuring
bond.  (Required for bonding mode 802.3ad).

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index f13a673..689ce1c 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -878,6 +878,15 @@ static void virtnet_vlan_rx_kill_vid(struct net_device *dev, u16 vid)
 		dev_warn(&dev->dev, "Failed to kill VLAN ID %d.\n", vid);
 }
 
+static int virtnet_get_settings(struct net_device *dev,
+	struct ethtool_cmd *ecmd)
+{
+	ethtool_cmd_speed_set(ecmd, SPEED_10000);
+	ecmd->duplex = DUPLEX_FULL;
+
+	return 0;
+}
+
 static void virtnet_get_ringparam(struct net_device *dev,
 				struct ethtool_ringparam *ring)
 {
@@ -891,6 +900,7 @@ static void virtnet_get_ringparam(struct net_device *dev,
 }
 
 static const struct ethtool_ops virtnet_ethtool_ops = {
+	.get_settings = virtnet_get_settings,
 	.get_link = ethtool_op_get_link,
 	.get_ringparam = virtnet_get_ringparam,
 };
